src/main/java/de/hysky/skyblocker/skyblock/slayers/SlayerManager.java

```diff
@@ -158,16 +171,21 @@ public static void getSlayerInfoOnJoin() {
 	 * {@link #findClosestMobEntity(EntityType, ArmorStandEntity)} could be modified and run more than once to ensure the correct entity is found.
 	 */
 	public static void checkSlayerBoss(ArmorStandEntity armorStand) {
-		if (slayerQuest == null || isBossSpawned() || !armorStand.hasCustomName() || !armorStand.isInRange(MinecraftClient.getInstance().player, 15)) return;
-		if (armorStand.getName().getString().contains(MinecraftClient.getInstance().getSession().getUsername())) {
+		if (slayerQuest == null || (isBossSpawned() && bossFight.boss != null) || !armorStand.hasCustomName()) return;
+		if (armorStand.getName().getString().contains(CLIENT.getSession().getUsername())) {
 			for (Entity otherArmorStands : getEntityArmorStands(armorStand, 1.5f)) {
 				Matcher matcher = SLAYER_PATTERN.matcher(otherArmorStands.getName().getString());
 				if (matcher.find()) {
+					if (bossFight.boss == null) {
```

Comment:
I don't think `bossFight` is guaranteed to be nonnull here? The scoreboard detection might break in rare cases so I'd just add a null check here.